src/appleseed/renderer/kernel/rendering/generic/genericsamplerenderer.cpp

```diff
@@ -253,6 +254,22 @@ namespace
                 if (!shading_point_ptr->hit_surface())
                     break;
 
+                const Material* material = shading_point_ptr->get_material();
+                if (material &&
```

Comment:
We now try to be explicit in pointer comparisons:

    if (material != nullptr &&